test/cases/api.sh

```diff
@@ -145,6 +145,25 @@ function instanceWaitSSH() {
   done
 }
 
+function instanceCheck() {
+  local _ssh="$1"
+
+  # Check if postgres is installed
+  $_ssh rpm -q postgresql
+
+  # Verify subscribe status
+  subscribe_org_id=$($_ssh sudo subscription-manager identity | grep 'org ID')
+  [[ "$subscribe_org_id" == "org ID: $API_TEST_SUBSCRIPTION_ORG_ID" ]]
+
+  # Remove dos2unix before install it
+  $_ssh rpm -q dos2unix && $_ssh sudo rpm -e dos2unix
```

Comment:
IMO you don't need to remove the rpm b/c if `rpm -q` returns non-zero the second part will not be executed anyway.  On top of that we have `set -e` enabled which should cause the entire script to exit with non-zero immediately.

Reply:
Hi @atodorov ,
I've tried it in my fedora33 that the "&&" can catch the previous exit code. Here is the logic:
if dos2unix is pre-installed, "rpm -q dos2unix" return 0, run "rpm -e dos2unix"
else if dos2unix is not pre-installed, "rpm -q dos2unix" return 1, then skip "rpm -e dos2unix" step and run "dnf -y install dos2unix"...
The only risk here I think is if "rpm -e dos2unix" failed the script will exit. But I think it's very rare to happen. And if the package was installed but failed to be removed, the next step "dnf install" will not be able to verify the package installation. 
Could you please double check that? Thanks!

Reply:
According to man `The shell does not exit if the command that fails is ... part of  any  command  exe‐cuted  in  a  &&  or || list except the command following the final && or ||,` so it looks like you are correct and I was wrong. 

rpm -e can indeed fail if there are other packages requiring dos2unix. But in this case we control what gets installed onto the system and which packages we use for testing purposes. It is unlikely that would fail and it is probably equally unlikely that we'll have to remove the package before installing it again. I don't have a strong opinion in any direction. 

@Gundersanne what do you think about  checking & removing the package before installing it vs. only checking that it's not installed and failing the entire script if it is ?

Reply:
Hm are we expecting the package to sometimes be installed? Or can we just pick a package that's never installed?

Plus wouldn't `dnf install $installed_package` fail anyway if the system isn't registered?

Reply:
Ah actually yes I think. If there's no repo, the yum install returns 1 no matter if the package is installed:
```
# yum install dos2unix
Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
# echo $?
1
```
So I think I should remove the package check before dnf install?

Reply:
Just removed the package pre-check. Please let me know if it's not a good solution. Thanks!

Reply:
Once we migrate to GitLab the runners will come with some epositories pre-configured sometimes.

If you want this to be a robust check then remove/disable  everything under /etc/yum.repos.d before subscribing but still keep the `rpm -q dos2unix` check.  That's a pre-condition and we want to fail if the pre-condition isn't met.

Reply:
Thanks @atodorov ! Since this PR is merged I opened a new issue #195 to track this, and will fix it soon.